salt/modules/cmdmod.py

```diff
@@ -2004,6 +2004,14 @@ def _cleanup_tempfile(path):
         # Backwards compatibility
         saltenv = __env__
 
+    if salt.utils.is_windows() and runas and cwd is None:
+        cwd = os.path.join(__opts__['cachedir'], 'wintmp')
```

Comment:
If it's truly necessary to apply directory permissions to an entire directory just for the sake of being able to one execute one file, I think that we should take the extra step of giving each script its own directory. What do you think about randomizing a directory name per-script and then caching the script there?